src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzer.cs

```diff
@@ -144,7 +144,13 @@ public override void Initialize(AnalysisContext analysisContext)
                 }
 
                 // Compute code metrics.
-                var computeTask = CodeAnalysisMetricData.ComputeAsync(compilationContext.Compilation, compilationContext.CancellationToken);
+                var inheritanceExcludedTypes = compilationContext.Options.GetInheritanceExcludedTypeNamesOption(CA1501Rule, compilationContext.Compilation, compilationContext.CancellationToken);
+
+                var metricsAnalysisContext = new CodeMetricsAnalysisContext(compilationContext.Compilation, compilationContext.CancellationToken)
+                {
+                    IsExcludedFromInheritanceCountFunc = namedType => inheritanceExcludedTypes.Contains(namedType)
```

Comment:
Just curious: any reason not to pass the delegate as the 3rd constructor argument?

Reply:
I can't say that's a good reason, I just thought that it wasn't scaling well to have all the extra func that we might add in the future as extra args.

Reply:
I am fine with adding it as extra ctor argument and wait until we see if we have some "scaling" problem. In the meantime this will be clearer.